R/forecastingdata_org.R

```diff
@@ -0,0 +1,380 @@
+## Functions to download data files from Monash Time Series Forecasting Archive, https://zenodo.org/communities/forecasting
+## Fore more details, please visit to https://forecastingdata.org/
+
+
+#' convert_tsf_to_tsibble function
+#'
+#' This function converts the contents in a .tsf file into a tsibble or a data.frame and returns it along with other meta-data of the dataset: frequency, horizon, whether the dataset contains missing values and whether the series have equal lengths.
+#' For more details, please refer to our paper: Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Archive.
+#'
+#' @param file .tsf file path
+#' @param value_column_name Any name that is preferred to have as the name of the column containing series values in the returning tsibble.
+#' @param key The name of the attribute that should be used as the key when creating the tsibble. If doesn't provide, a data frame will be returned instead of a tsibble.
+#' @param index The name of the time attribute that should be used as the index when creating the tsibble. If doesn't provide, it will search for a valid index. When no valid index found, a data frame will be returned instead of a tsibble.
+#'
+#' @return This function returns a tsibble or a data.frame based on the validity of given key and index values.
+#' @importFrom utils head tail
+#' 
+#' @references 
+#' Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Repository. \url{https://forecastingdata.org/}
+#' 
+#' Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Archive.
+#' 
+#' @export
+convert_tsf_to_tsibble <-   function(file, value_column_name = "series_value", key = NULL, index = NULL){
+  options(pillar.sigfig = 7)
```

Comment:
You shouldn't set user options in a package without restoring them afterwards. I don't know why the displayed sigfig needs changing in this function.
More details here: https://r-pkgs.org/r.html#code-r-landscape

Reply:
Fixed as suggested. Removed options.